File: ios/Sources/batteryPlugin/batteryPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(batteryPlugin)
public class batteryPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "batteryPlugin"
    public let jsName = "battery"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = battery()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}

File: android/src/main/java/com/testing/plugin/battery/batteryPlugin.java
package com.testing.plugin.battery;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "battery")
public class batteryPlugin extends Plugin {

    private battery implementation = new battery();

    @PluginMethod
    public void getBatteryLevel(PluginCall call) {
      JSObject ret = new JSObject();
      ret.put("level",100)
      call.resolve(ret);
    }
  
}
